src/app/server-cluster/ServerClusterInterface.h

```diff
@@ -153,21 +172,43 @@ class ServerClusterInterface
     ///   - if a value is returned (not nullopt) then the handler response MUST NOT be filled. The caller
     ///     will then issue `handler->AddStatus(request.path, <return_value>->GetStatusCode())`. This is a
     ///     convenience to make writing Invoke calls easier.
+    ///
+    /// Precondition:
+    ///   - `request.path` cluster part MUST be a path that is part of `GetPaths` returned values.
     virtual std::optional<DataModel::ActionReturnStatus>
     InvokeCommand(const DataModel::InvokeRequest & request, chip::TLV::TLVReader & input_arguments, CommandHandler * handler) = 0;
 
     /// Retrieves a list of commands accepted by this cluster.
     ///
     /// Returning `CHIP_NO_ERROR` without adding anything to the `builder` list is expected
     /// if no commands are supported by the cluster.
+    ///
+    /// Precondition:
+    ///   - `path` cluster part MUST be a path that is part of `GetPaths` returned values.
     virtual CHIP_ERROR AcceptedCommands(const ConcreteClusterPath & path,
                                         DataModel::ListBuilder<DataModel::AcceptedCommandEntry> & builder) = 0;
 
     /// Retrieves a list of commands generated by this cluster.
     ///
     /// Returning `CHIP_NO_ERROR` without adding anything to the `builder` list is expected
     /// if no commands are generated by processing accepted commands.
+    ///
+    /// Precondition:
+    ///   - `path` cluster part MUST be a path that is part of `GetPaths` returned values.
     virtual CHIP_ERROR GeneratedCommands(const ConcreteClusterPath & path, DataModel::ListBuilder<CommandId> & builder) = 0;
+
+    /// Returns if `GetPaths` contains the given path
+    bool PathsContains(const ConcreteClusterPath & path)
+    {
+        for (auto & myPath : GetPaths())
```

Comment:
const auto

Reply:
Done. Although type is identical either way ...